main/mio.c

```diff
@@ -990,7 +990,10 @@ char *mio_gets (MIO *mio, char *s, size_t size)
 				mio->impl.mem.pos++;
 				if (s[i] == '\n')
 				{
-					i++;
+					if (i > 0 && s[i-1] == '\r')
+						s[i-1] = '\n';
+					else
+						i++;
```

Comment:
```
+               else if (s[i] == '\r')
+               {
+                   i++;
```

This will require reading something into s[i] before the '\n' check happens but yeah, my version didn't treat \r only as a newline.

```
+                   if (s[i] == '\n')
+                       i++;
+                   break;
+               }
```